File: pdu/python/hako_msgs/pdu_cdr_conv_ShareObjectOwner.py
from .pdu_pytype_ShareObjectOwner import ShareObjectOwner
from ..pdu_cdr_runtime import CdrReader, CdrWriter

from ..geometry_msgs.pdu_cdr_conv_Twist import *
from ..geometry_msgs.pdu_cdr_conv_Vector3 import *



def py_to_cdr_body_ShareObjectOwner(writer: CdrWriter, src: ShareObjectOwner):
    writer.write_string(src.object_name)
    writer.write_uint32(src.owner_id)
    writer.write_uint64(src.last_update)
    py_to_cdr_body_Twist(writer, src.pos)


def cdr_body_to_py_ShareObjectOwner(reader: CdrReader, dst: ShareObjectOwner):
    dst.object_name = reader.read_string()
    dst.owner_id = reader.read_uint32()
    dst.last_update = reader.read_uint64()
    cdr_body_to_py_Twist(reader, dst.pos)
    return dst


def py_to_cdr_ShareObjectOwner(src: ShareObjectOwner) -> bytes:
    writer = CdrWriter()
    writer.write_encapsulation()
    py_to_cdr_body_ShareObjectOwner(writer, src)
    return writer.bytes()


def cdr_to_py_ShareObjectOwner(cdr_payload) -> ShareObjectOwner:
    reader = CdrReader(cdr_payload)
    reader.read_encapsulation()
    dst = ShareObjectOwner()
    return cdr_body_to_py_ShareObjectOwner(reader, dst)

File: pdu/javascript/hako_msgs/pdu_cdr_conv_ShareObjectOwner.js
import { ShareObjectOwner } from './pdu_jstype_ShareObjectOwner.js';
import { PduCdrWriter, PduCdrReader } from '../pdu_cdr_runtime.js';
import { Twist } from '../geometry_msgs/pdu_jstype_Twist.js';
import { PduTwistConverter } from '../geometry_msgs/pdu_cdr_conv_Twist.js';
import { Vector3 } from '../geometry_msgs/pdu_jstype_Vector3.js';
import { PduVector3Converter } from '../geometry_msgs/pdu_cdr_conv_Vector3.js';


export class PduShareObjectOwnerConverter {
    /**
     * @param {PduCdrWriter} writer
     * @param { ShareObjectOwner } src
     */
    static to_cdr_body(writer, src) {
        writer.write_string(src.object_name);
        writer.write_uint32(src.owner_id);
        writer.write_uint64(src.last_update);
        PduTwistConverter.to_cdr_body(writer, src.pos);
    }

    /**
     * @param {PduCdrReader} reader
     * @param { ShareObjectOwner } dst
     * @returns { ShareObjectOwner }
     */
    static cdr_body_to_js(reader, dst) {
        dst.object_name = reader.read_string();
        dst.owner_id = reader.read_uint32();
        dst.last_update = reader.read_uint64();
        PduTwistConverter.cdr_body_to_js(reader, dst.pos);
        return dst;
    }

    /**
     * @param { ShareObjectOwner } src
     * @returns {ArrayBuffer}
     */
    static to_cdr(src) {
        const writer = new PduCdrWriter();
        writer.write_encapsulation();
        this.to_cdr_body(writer, src);
        return writer.get_buf();
    }

    /**
     * @param {ArrayBuffer|ArrayBufferView} cdrPayload
     * @returns { ShareObjectOwner }
     */
    static from_cdr(cdrPayload) {
        const reader = new PduCdrReader(cdrPayload);
        reader.read_encapsulation();
        return this.cdr_body_to_js(reader, new ShareObjectOwner());
    }
}
